File: src/test/samples/validSamples/validSample.cs
#region FirstRegion
int x = 42;
#endregion

#region Second Region
class MyClass {
    #region    InnerRegion  
    readonly int x = 42;
    int y = 100;
    protected int z = 200;
    private void MyMethod() {}
    #endregion   ends  InnerRegion    

    #region
    static void MyMethod2() {}
    #endregion  
}
#endregion

File: src/test/samples/validSamples/validSample.rs
fn main() {
    // #region FirstRegion
    let x = 42;
    //#endregion

    //    #region Second Region
    struct MyClass;

    impl MyClass {
        // #region InnerRegion
        fn my_method(&self) {}
        //   #endregion   ends InnerRegion

        // #region
        fn my_method2(&self) {}
        //#endregion
    }
    //  #endregion
}
